### PR TITLE
BAMBK8S-129: Bamboo pre-seeding configuration

### PR DESCRIPTION
**Pull request by tarka**
Note: This will not fully work until 8.1 is released. Additionally it will currently fail CI due to missing stashed vars in our Bamboo CI.

I've added the optional injection of the necessary secrets to `helm_install`, however I'm not sure if this is appropriate, or if we should just manually add them to the CI cluster/NS (as is done with DB credentials ATM).

### CODE REVIEW

File: src/test/scripts/helm_install.sh [hunk 1]
```diff
@@ -227,6 +227,30 @@ package_functest_helm_chart() {
   helm package "$FUNCTEST_CHART_PATH" --destination "$HELM_PACKAGE_DIR"
 }
 
+install_secrets() {
+    if [[ -n "${BAMBOO_LICENSE}" ]]; then
```
Comment (nanux) [inline src/test/scripts/helm_install.sh:231]: Is this just for testing locally? We can install the secrets to our CI cluster if necessary.

File: src/main/charts/bamboo/values.yaml [hunk 2]
```diff
@@ -340,7 +340,67 @@ ingress:
...
+
+    # -- The key (default 'licenseKey') in the Secret used to store the license information
+    #
+    licenseKey: license
+
+  # -- The admin user configuration and credentials to provide to
+  # Bamboo (optional). If provided this will skip the configuration
+  # screen on first run.
+  #
+  adminUser:
```
Comment (nanux) [inline src/main/charts/bamboo/values.yaml:382]: I think we should make it consistent with Bitbucket and call it `sysadminCredentials`
https://github.com/atlassian/data-center-helm-charts/blob/main/src/main/charts/bitbucket/values.yaml#L528
I don't think `sysadminCredentials` is perfect but it is an established term.

File: src/main/charts/bamboo/values.yaml [hunk 3]
```diff
@@ -340,7 +340,67 @@ ingress:
...
+    #
+    usernameSecretKey: username
+
+    # -- The key (default `password`) in the Secret used to store the admin password
+    #
+    passwordSecretKey: password
+
+    # -- The key (default `fullname`) in the Secret used to store the admin full name
+    #
+    fullnameSecretKey: fullname
```
Comment (nanux) [inline src/main/charts/bamboo/values.yaml:398]: For consistency with BB this should be `displayNameSecretKey`

File: src/main/charts/bamboo/values.yaml [hunk 4]
```diff
@@ -340,7 +340,67 @@ ingress:
...
+    
+    # -- The key (default `email`) in the Secret used to store the admin email
+    #
+    emailSecretKey: email
```
Comment (nanux) [inline src/main/charts/bamboo/values.yaml:402]: Again, BB uses `emailAddressSecretKey`

File: src/main/charts/bamboo/values.yaml [hunk 5]
```diff
@@ -340,7 +340,67 @@ ingress:
...
+  #
+  license:
+
+    # -- The secret that contains the license information
+    #
+    secretName:
+
+    # -- The key (default 'licenseKey') in the Secret used to store the license information
+    #
+    licenseKey: license
```
Comment (nanux) [inline src/main/charts/bamboo/values.yaml:376]:
```suggestion
    secretKey: license
```

The other products are using `secretKey` for the license.
https://github.com/atlassian/data-center-helm-charts/blob/main/src/main/charts/jira/values.yaml#L470

File: src/main/charts/bamboo/values.yaml [hunk 6]
```diff
@@ -340,7 +340,67 @@ ingress:
...
+
+  # -- Override the server/agent broker URL; this is optional.
+  brokerUrl:
+
+  # The security token with which the agent will authenticate to the Bamboo server
+  # Additional details here:
+  # https://confluence.atlassian.com/bamboo/agent-authentication-289277196.html#Agentauthentication-SecuritytokenverificationSecuritytokenverification
+  #
+  securityToken:
+    # -- The name of the K8s Secret that contains the security token. When specified the token
```
Comment (nanux) [inline src/main/charts/bamboo/values.yaml:352]: Missing empty line

File: src/test/config/bamboo/values.yaml [hunk 7]
```diff
@@ -3,8 +3,19 @@
 bamboo:
   clustering:
     enabled: ${clustering.enabled}
+
+  securityToken:
```
Comment (nanux) [inline src/test/config/bamboo/values.yaml:7]: Good idea to add it to the expected output

**Review by nanux — CHANGES_REQUESTED** [review]

File: src/test/scripts/helm_install.sh (hunk 1 above)
Comment (tarka) [reply src/test/scripts/helm_install.sh:231]: I mention this in the description above; I'm not sure if this is appropriate, I'm happy to remove it. However we should have somewhere where this is automated/documented as a prerequisite for the CI cluster.

**Review by nanux — APPROVED** [review]

File: src/main/charts/bamboo/values.yaml [hunk 8]
```diff
@@ -340,7 +340,94 @@ ingress:
 # Bamboo configuration 
 #
 bamboo:
-  
+
+  # -- Supply the base URL for the installation.
```
Comment (errcode1202) [inline src/main/charts/bamboo/values.yaml:344]: For consistency can we include an additional `#` under this line

File: src/main/charts/bamboo/values.yaml [hunk 9]
```diff
@@ -340,7 +340,94 @@ ingress:
...
+  baseUrl:
+
+  # -- Override the server/agent broker URL; this is optional.
```
Comment (errcode1202) [inline src/main/charts/bamboo/values.yaml:347]: For consistency can we include an additional `#` under this line

File: src/main/charts/bamboo/values.yaml [hunk 10]
```diff
@@ -340,7 +340,94 @@ ingress:
...
+    # 'kubectl create secret generic <secret-name> --from-literal=security-token=<security token>'
+    # https://kubernetes.io/docs/concepts/configuration/secret/#opaque-secrets
+    #
+    secretName:
+
+    # -- The key (default `secretKey`) in the Secret used to store the Bamboo shared key.
+    #
+    secretKey: security-token
+
+  # -- Whether to disable agent authentication. Default is false.
```
[the remark this answers is not in the document]
Comment (errcode1202) [reply src/main/charts/bamboo/values.yaml:368]: i.e. useful for auto approval of agents

File: src/main/charts/bamboo/values.yaml [hunk 11]
```diff
@@ -340,7 +340,94 @@ ingress:
...
 bamboo:
-  
+
+  # -- Supply the base URL for the installation.
+  baseUrl:
+
+  # -- Override the server/agent broker URL; this is optional.
+  brokerUrl:
+
+  # The security token with which the agent will authenticate to the Bamboo server
```
Comment (errcode1202) [inline src/main/charts/bamboo/values.yaml:350]: Do we need to include details on the expected format of the token? Does it need to be of a certain length/format/encoding?

File: src/main/charts/bamboo/values.yaml [hunk 12]
```diff
@@ -340,7 +340,94 @@ ingress:
...
+    
+    # -- The key in the Kubernetes Secret that contains the sysadmin display name
+    #
+    displayNameSecretKey: displayName
+    
+    # -- The key in the Kubernetes Secret that contains the sysadmin email address
+    #
+    emailAddressSecretKey: emailAddress
+
+  # -- Bamboo can optionally import an existing exported dataset on
```
Comment (errcode1202) [inline src/main/charts/bamboo/values.yaml:413]: Can we also include a reference to the official docs on this behaviour i.e.

- https://confluence.atlassian.com/bamboo/exporting-data-for-backup-289277255.html
- https://confluence.atlassian.com/bamboo/importing-data-from-backup-289277260.html

File: src/main/charts/bamboo/values.yaml [hunk 13]
```diff
@@ -340,7 +340,94 @@ ingress:
...
+  # first-run. These optional values can configure the import file or
+  # skip this stage entirely.
+  #
+  import:
+
+    # -- Import type. Valid values are 'clean' (for a fresh install)
```
Comment (errcode1202) [inline src/main/charts/bamboo/values.yaml:419]: not sure about the word `fresh` it's a bit of a colloquialism. Maybe `new` is better?

**Review by errcode1202 — CHANGES_REQUESTED** [review]

**Review by errcode1202 — APPROVED** [review]

File: src/main/charts/bamboo/values.yaml [hunk 14]
```diff
@@ -340,7 +340,105 @@ ingress:
...
+  # -- Supply the base URL for the installation.
+  #
+  baseUrl:
+
+  # -- Override the server/agent broker URL; this is optional.
+  #
+  brokerUrl:
+
+  # The security token with which the agent will authenticate to the
+  # Bamboo server. The token should be set to a 40-character
```
Comment (tarka) [inline src/main/charts/bamboo/values.yaml:353]: @achystoprudov Can you confirm this is the correct format?

Comment (achystoprudov) [reply src/main/charts/bamboo/values.yaml:353]: @tarka , at Bamboo JavaDocs we wrote
```
*  Sets token value for Agent authentication (if defined bamboo won't generate new token).
 *  Token should contain only hexadecimal characters and have 40 characters length.
*  <li>Usage: <ul><li>env : bamboo.setup.remote.agent.security.token.value="ABCDEF12345678901234abcdef0123456789abcd"
*                  <li>java: -Dbamboo.setup.remote.agent.security.token.value=ABCDEF12345678901234abcdef0123456789abcd
*             </ul>
*  </ul>
```

**Pull request merged**